plugins/net_plugin/net_plugin.cpp

```diff
@@ -4670,56 +4640,75 @@ namespace eosio {
          return "invalid peer address";
       }
 
-      std::lock_guard g( connections_mtx );
-      if( find_connection_i( peer_address ) )
-         return "already connected";
+      {
+         std::lock_guard g( connections_mtx );
```

Comment:
If `connections_mtx` is to remain a `shared_mutex` (which I'm not sure is a great idea), then this probably should be `std::shared_lock`.  Same for `connections_manager::number_connections()` and the unique_lock in `connections_manager::connect_supplied_peers`.